plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/utils/JdkFetcher.java

```diff
@@ -0,0 +1,270 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A utility class for fetching the latest JDK releases from the Adoptium GitHub repository.
+ */
+@SuppressFBWarnings(value = "PATH_TRAVERSAL_IN", justification = "false, positive")
+public class JdkFetcher {
+    private static final Logger LOG = LoggerFactory.getLogger(JdkFetcher.class);
+
+    private static final String GITHUB_API_URL = "https://api.github.com/repos/adoptium";
+    private static final String DOWNLOAD_DIR = System.getProperty("user.home") + "/.jdks/";
+
+    /**
+     * Gets the path to the JDK directory for the specified JDK version. If the JDK is not already downloaded,
+     * it triggers the download and setup process.
+     *
+     * @param jdkVersion The version of the JDK (e.g., "8").
+     * @return The path to the JDK directory.
+     * @throws IOException          If an I/O error occurs.
+     * @throws InterruptedException If the operation is interrupted.
+     */
+    public Path getJdkPath(String jdkVersion) throws IOException, InterruptedException {
+        Path jdkPath = Paths.get(DOWNLOAD_DIR, "plugin-modernizer-jdk-" + jdkVersion);
+        if (Files.notExists(jdkPath)) {
+            downloadAndSetupJdk(jdkVersion);
+        }
+        return jdkPath;
+    }
+
+    /**
+     * Downloads and extracts the JDK for the specified version. The method determines the appropriate extraction
+     * method based on the operating system.
+     *
+     * @param jdkVersion The version of the JDK (e.g., "8").
+     * @throws IOException          If an I/O error occurs.
+     * @throws InterruptedException If the operation is interrupted.
+     */
+    private void downloadAndSetupJdk(String jdkVersion) throws IOException, InterruptedException {
+        LOG.info("Downloading the jdk...");
+        Path downloadedFile = downloadJdk(jdkVersion);
+        LOG.info("Download Successful");
+        String os = getOSName();
+        LOG.info("Extracting...");
+        Path extractionDir = Paths.get(DOWNLOAD_DIR, "plugin-modernizer-jdk-" + jdkVersion);
+        Files.createDirectories(extractionDir);
+        if (os.contains("windows")) {
+            extractZip(downloadedFile, extractionDir);
+        } else if (os.contains("linux") || os.contains("mac")) {
+            extractTarGz(downloadedFile, extractionDir);
+        }
+        LOG.info("Extraction successful");
+    }
+
+    /**
+     * Downloads the JDK for the specified version from the appropriate URL and saves it to the local directory.
+     *
+     * @param jdkVersion The version of the JDK (e.g., "8").
+     * @return The path to the downloaded JDK file.
+     * @throws IOException          If an I/O error occurs.
+     * @throws InterruptedException If the operation is interrupted.
+     */
+    @SuppressFBWarnings(value = "URLCONNECTION_SSRF_FD", justification = "false positive")
+    private Path downloadJdk(String jdkVersion) throws IOException, InterruptedException {
```

Comment:
What do you think about an Enum ? We pass always the major version, never the minor version or path.

 Adding a new JDK could be just adding a new value on the enum